In [21]:
!pip install pandas


[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


This is the dataset from Kaggle Dataset 

In [26]:
import pandas as pd

restaurants_df=pd.read_csv("data/Grab SG Restaurants.csv")

restaurants_df.head()


id_source             name                     address    country  \
0         SGDD01085    Texas Chicken    Texas Chicken - VivoCity  Singapore   
1  4-CYTDLPUJEP53N6           Subway           Subway - Vivocity  Singapore   
2         SGDD05605         LiHO Tea        LiHO Tea - Vivo City  Singapore   
3         SGDD07548         Wingstop         Wingstop - VivoCity  Singapore   
4  4-CY2ZLZLTT2EDJ2  Le Shrimp Ramen  Le Shrimp Ramen - Vivocity  Singapore   

                                             cuisine currency  delivery_cost  \
0  ["Western", "Fast Food", "Halal", "Chicken", "...      SGD          570.0   
1  ["Healthy", "Sandwiches", "Fast Food", "Halal"...      SGD          570.0   
2  ["Same Prices In-Store", "Beverages", "Local",...      SGD          560.0   
3  ["Halal", "Chicken", "Fast Food", "Burger", "W...      SGD          560.0   
4          ["Japanese", "Ramen", "Noodles", "Asian"]      SGD          570.0   

        lat         lon                                      opening_hours  \
0  1.264792  103.822103  {"open": true, "displayedHours": "10:30-21:30"...   
1  1.264741  103.822072  {"open": true, "displayedHours": "12:00-19:00"...   
2  1.264309  103.821369  {"open": true, "displayedHours": "11:30-20:45"...   
3  1.263527  103.821712  {"open": true, "displayedHours": "11:00-21:00"...   
4  1.264810  103.822554  {"open": true, "displayedHours": "10:30-21:00"...   

                                           image_url  radius  rating  \
0  https://d1sag4ddilekf6.cloudfront.net/compress...    5000     3.8   
1  https://d1sag4ddilekf6.cloudfront.net/compress...    3000     4.2   
2  https://d1sag4ddilekf6.cloudfront.net/compress...    5000     4.5   
3  https://d1sag4ddilekf6.cloudfront.net/compress...    5000     4.4   
4  https://d1sag4ddilekf6.cloudfront.net/compress...    5000     4.6   

   reviews_nr   delivery_options  \
0      1591.0      ONLY_DELIVERY   
1       197.0      ONLY_DELIVERY   
2       353.0  DELIVERY_TAKEAWAY   
3      1391.0      ONLY_DELIVERY   
4       309.0  DELIVERY_TAKEAWAY   

                                               promo loc_type delivery_by  \
0  Use 'TEATIME' for FREE delivery between 3PM to...     FOOD        GRAB   
1  Use 'TEATIME' for FREE delivery between 3PM to...     FOOD        GRAB   
2  Use 'TEATIME' for FREE delivery between 3PM to...     FOOD        GRAB   
3  Use 'TEATIME' for FREE delivery between 3PM to...     FOOD        GRAB   
4  Use 'TEATIME' for FREE delivery between 3PM to...     FOOD        GRAB   

   delivery_time  
0           43.0  
1           42.0  
2           41.0  
3           45.0  
4           44.0

## Exploratory Data Analysis

Simple exploratory Data Analysis to understand the data

In [27]:
restaurants_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16136 entries, 0 to 16135
Data columns (total 19 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_source         16136 non-null  object 
 1   name              16130 non-null  object 
 2   address           16136 non-null  object 
 3   country           16136 non-null  object 
 4   cuisine           16134 non-null  object 
 5   currency          16117 non-null  object 
 6   delivery_cost     16085 non-null  float64
 7   lat               16136 non-null  float64
 8   lon               16136 non-null  float64
 9   opening_hours     16136 non-null  object 
 10  image_url         16125 non-null  object 
 11  radius            16136 non-null  int64  
 12  rating            14004 non-null  float64
 13  reviews_nr        13602 non-null  float64
 14  delivery_options  16136 non-null  object 
 15  promo             8777 non-null   object 
 16  loc_type          16136 non-null  object

So the dataset of 16136 rows of grab restaurants

In [28]:
restaurants_df.describe()

delivery_cost           lat           lon        radius        rating  \
count   16085.000000  16136.000000  16136.000000  16136.000000  14004.000000   
mean      984.926640      1.338491    103.843448   6871.493183      4.238346   
std       656.544752      0.045269      0.062123   6170.013565      0.619217   
min       199.000000      1.230454    103.625677    578.000000     -1.000000   
25%       450.000000      1.304839    103.806196   4999.000000      4.000000   
50%       680.000000      1.332154    103.846004   5000.000000      4.300000   
75%      1610.000000      1.362576    103.885674   5000.000000      4.600000   
max     14000.000000      1.469011    103.997253  25000.000000      5.000000   

         reviews_nr  delivery_time  
count  13602.000000   15953.000000  
mean     343.591604      37.552811  
std      846.893054      15.728705  
min        1.000000      15.000000  
25%       16.000000      28.000000  
50%       74.000000      34.000000  
75%      303.000000      42.000000  
max    21113.000000     169.000000

As total rows should be 16136 you can deduce that are null values in delivery_cost, rating, reviews_nr and delivery time. 

I am concern on delivery_cost and delivery_time column as this shows discrepancies on the data.



In [24]:


#unique of loc_type 2 types food mart
#main_df['loc_type'].unique()

#2 type grab, merchant
#main_df['delivery_by'].unique()

# #seperate after the hyphen into a new category
#main_df['address'].unique()

# #currency there are 2 type SGD and NAN
# main_df['currency'].unique()

main_df['delivery_options'].unique()

#main_df['promo'].unique()

array(['ONLY_DELIVERY', 'DELIVERY_TAKEAWAY', 'DELIVERY_DINEIN',
       'DELIVERY_TAKEAWAY_DINEIN'], dtype=object)

In [ ]:
restaurant_df = main_df[['Id_source', 'name', 'address', 'rating', 'reviews_nr', 'opening_hr', 'delivery_time']]


In [9]:
#creating a new dataframe with only the cuisine category
cuisine_df=pd.DataFrame(columns=['source_id','cuisine'])

#iterate rows through each row of the original main_df

for idx,row in main_df.iterrows():
    #extract cuisine type
    cuisines = row['cuisine']

    #iterate through each cuisine type 
    for cuisine in cuisines:
        #append the cuisine type and source column ID to the newly created cuisine_df
        cuisine_df=cuisine_df.append({'source_id': row['source_id','cuisine': cuisine]}, ignore_index=True)

cuisine_df.head()

AttributeError: 'DataFrame' object has no attribute 'append'